sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -115,55 +114,24 @@ def test_hasher_zeros():
 def test_hasher_alternate_sign():
     X = [list("Thequickbrownfoxjumped")]
 
-    Xt = FeatureHasher(alternate_sign=True, non_negative=False,
+    Xt = FeatureHasher(alternate_sign=True,
                        input_type='string').fit_transform(X)
     assert Xt.data.min() < 0 and Xt.data.max() > 0
 
-    Xt = FeatureHasher(alternate_sign=True, non_negative=True,
-                       input_type='string').fit_transform(X)
-    assert Xt.data.min() > 0
-
-    Xt = FeatureHasher(alternate_sign=False, non_negative=True,
+    Xt = FeatureHasher(alternate_sign=False,
                        input_type='string').fit_transform(X)
     assert Xt.data.min() > 0
-    Xt_2 = FeatureHasher(alternate_sign=False, non_negative=False,
-                         input_type='string').fit_transform(X)
-    # With initially positive features, the non_negative option should
-    # have no impact when alternate_sign=False
-    assert_array_equal(Xt.data, Xt_2.data)
 
 
-@ignore_warnings(category=DeprecationWarning)
 def test_hash_collisions():
     X = [list("Thequickbrownfoxjumped")]
 
-    Xt = FeatureHasher(alternate_sign=True, non_negative=False,
-                       n_features=1, input_type='string').fit_transform(X)
+    Xt = FeatureHasher(alternate_sign=True, n_features=1,
+                       input_type='string').fit_transform(X)
     # check that some of the hashed tokens are added
     # with an opposite sign and cancel out
     assert abs(Xt.data[0]) < len(X[0])
 
-    Xt = FeatureHasher(alternate_sign=True, non_negative=True,
-                       n_features=1, input_type='string').fit_transform(X)
-    assert abs(Xt.data[0]) < len(X[0])
-
-    Xt = FeatureHasher(alternate_sign=False, non_negative=True,
-                       n_features=1, input_type='string').fit_transform(X)
+    Xt = FeatureHasher(alternate_sign=False, n_features=1,
+                       input_type='string').fit_transform(X)
     assert Xt.data[0] == len(X[0])
-
-
-@ignore_warnings(category=DeprecationWarning)
-def test_hasher_negative():
-    X = [{"foo": 2, "bar": -4, "baz": -1}.items()]
-    Xt = FeatureHasher(alternate_sign=False, non_negative=False,
-                       input_type="pair").fit_transform(X)
-    assert Xt.data.min() < 0 and Xt.data.max() > 0
-    Xt = FeatureHasher(alternate_sign=False, non_negative=True,
-                       input_type="pair").fit_transform(X)
-    assert Xt.data.min() > 0
-    Xt = FeatureHasher(alternate_sign=True, non_negative=False,
-                       input_type="pair").fit_transform(X)
-    assert Xt.data.min() < 0 and Xt.data.max() > 0
-    Xt = FeatureHasher(alternate_sign=True, non_negative=True,
-                       input_type="pair").fit_transform(X)
-    assert Xt.data.min() > 0
```

Comment:
I think it doesn't hurt to keep this test (but remove the `non_negative=False` parts), unless you think it's redundant?

Reply:
This test without the `non-negative` argument is the exact same test as `test_hasher_alter_sign`.